app/assets/javascripts/host_edit.js

```diff
@@ -2,6 +2,43 @@ $(document).on('ContentLoad', function(){onHostEditLoad()});
 $(document).on('AddedClass', function(event, link){load_puppet_class_parameters(link)});
 $(document).on('click', '#params-tab', function() { resizeTextareas($('#params')); });
 
+LookupValueController = function(item) {
+  var self = this; //because JS' this confuses me
```

Comment:
I understand you like ruby better, but trying to force JS look like ruby isn't a good idea in my opinion. 
After you finish working on this code other developers will need to work with it and they will be looking for JS conventions (e.g. `prototype`) when working on a JS file.